src/Kafka/Consumer/Callbacks.hs

```diff
@@ -94,3 +97,12 @@ assign (KafkaConsumer (Kafka k) _) ps =
                 else toNativeTopicPartitionList ps
         er = KafkaResponseError <$> (pl >>= rdKafkaAssign k)
     in kafkaErrorToMaybe <$> er
+
+-- | Assigns specified partitions to a current consumer.
+-- Assigning an empty list means unassigning from all partitions that are currently assigned.
+assign' :: KafkaConsumer -> RdKafkaTopicPartitionListTPtr -> IO (Maybe KafkaError)
+assign' (KafkaConsumer (Kafka k) _) pls = do
+    er <- KafkaResponseError <$> rdKafkaAssign k pls
```

Comment:
Since `NO_ERROR` is not an error please use this one instead:
https://github.com/haskell-works/hw-kafka-client/blob/master/src/Kafka/Internal/Shared.hs#L64